docs/blocks/index.md

```diff
@@ -92,6 +92,73 @@ Directly using `Block` only allows for very basic blocks. If you want to add fun
 
 If you want to make a block that has different variants (think a slab that has a bottom, top, and double variant), you should use [blockstates]. And finally, if you want a block that stores additional data (think a chest that stores its inventory), a [block entity][blockentities] should be used. The rule of thumb here is that if you have a finite and reasonably small amount of states (= a few hundred states at most), use blockstates, and if you have an infinite or near-infinite amount of states, use a block entity.
 
+#### Block Types
+
+Block types are [`MapCodec`s][codec] used to serialize and deserialize a block object. This `MapCodec` is set via `BlockBehaviour#codec` and [registered][registration] to the block type registry. Currently, its only use is when the block list report is being generated. A block type should be created once for a class and its subclass if its constructor takes in more than just the properties and the logic within the class is substantially different from its super. For example, `FlowerBlock#CODEC` represents the block type for most flowers. Since its subclass, `WitherRoseBlock`, provides a difference in logic, a separate block type is created.
```

Comment:
This wording is weird. If a given block class has a subclass, then a dedicated block type should exist for it in any case, otherwise the subclass is useless

Reply:
Better?

Reply:
Yes